### DISCUSSION TITLE
Why partitions cannot be loaded into memory？

### DISCUSSION THREAD

**Discussion by upInTheWinds** [milvus-io/milvus#23807] [Q&A and General discussion]
In milvus under 2.2.6, we tried to use the go API to load the partition for the first time, but the program was blocked all the time. The log of querycoord is as follows. What is the problem?
[quertcoordLogs.txt](https://github.com/milvus-io/milvus/files/11352270/quertcoordLogs.txt)

**Comment by yhmo** [comment]
I just test on my local, create a collection->create a partition->insert some rows into the partition->load the partition, seems it works well.
The log shows the loading task is begin, but the log doesn't reveal why the loading task is not finished. Could you attach more log about the query nodes.

**Reply by upInTheWinds** [reply]
My testing process is create a collection->create a partition->insert some rows into the partition->build index -> load the partition.The logs of querynode and querycoord are as follows. 
[querycoordLogfs.txt](https://github.com/milvus-io/milvus/files/11392295/querycoordLogfs.txt)
[querynodeLogs.txt](https://github.com/milvus-io/milvus/files/11392296/querynodeLogs.txt)

**Reply by yhmo** [reply]
The log shows:
`[2023/05/04 00:52:27.313 +00:00] [INFO] [observers/collection_observer.go:283] ["partition load status updated"] [collectionID=441230618091323885] [partitionID=441230618091324025] [loadPercentage=100] [partitionStatus=2]`

The loading process is successfully done, partitionStatus=2 means the partition is "loaded". It seems no problem.

I also tried create a collection->create a partition->insert some rows into the partition->build index -> load the partition, no problem here:
```
if __name__ == '__main__':
    connections.connect(host='127.0.0.1', port="19530")

    id_field = "id"
    vector_field = "vec"
    dim = 128
    collection_name = "demo"

    if utility.has_collection(collection_name):
        utility.drop_collection(collection_name)

    field1 = FieldSchema(name=id_field, dtype=DataType.INT64, is_primary=True, auto_id=True)
    field2 = FieldSchema(name=vector_field, dtype=DataType.FLOAT_VECTOR, dim=dim)
    schema = CollectionSchema(fields=[field1, field2])
    collection = Collection(name=collection_name, schema=schema, shards_num=1)
    print("create collection")
    collection.create_partition(partition_name="p1")
    print("create partition")

    data = [
        [[random.random() for _ in range(dim)] for _ in range(10000)],
    ]
    collection.insert(data, partition_name="p1")
    print("insert done")

    index_param = {"index_type": "HNSW", "metric_type": "L2", "params": {"M": 48, "efConstruction": 500}}
    collection.create_index(vector_field, index_param)
    print("index done")

    collection.load(partition_names=["p1"])
    print("load partitiondone")

    search_param = {
        "data": [[random.random() for _ in range(dim)]],
        "anns_field": vector_field,
        "param": {"metric_type": "L2", "params": {"ef": 16}},
        "limit": 3,
    }
    results = collection.search(**search_param)
    print("search done")
    print(results)
```
